sip-ri/src/main/java/gov/nist/javax/sip/parser/extensions/TargetDialogParser.java

```diff
@@ -25,49 +27,18 @@ public SIPHeader parse() throws ParseException {
             headerName(TokenTypes.TARGET_DIALOG);
             this.lexer.SPorHT();
 
-            String callId;
-            char firstChar = this.lexer.lookAhead(0);
-            if (firstChar == '<') {
-                // Enclosed in <> (quoted string)
-                callId = this.lexer.quotedString();
-                this.lexer.match('>');
-            } else {
-                // Not enclosed in <> (SIP URI)
-                callId = this.lexer.byteStringNoSemicolon();
-
-                // Check for unexpected characters (excluding whitespace)
-                if (callId.indexOf('<') >= 0 || callId.indexOf('>') >= 0) {
-                    throw new ParseException("Unexpected characters in Call-Id (<>): " + callId, this.lexer.getPtr());
-                }
-            }
-
+            String callId = this.lexer.byteStringNoSemicolon();
             targetDialog.setCallId(callId.trim());
 
-            // Check for optional parameters (local-tag and/or remote-tag)
-            if (this.lexer.lookAhead(0) == ';') {
-                this.lexer.match(';');
-
-                // Parse local-tag (if present)
-                if (this.lexer.lookAhead(0) == 'l') {
-                    this.lexer.match('l');
-                    this.lexer.match('=');
-                    targetDialog.setLocalTag(this.lexer.quotedString());
-
-                    // Optional semicolon after local-tag (consume if present)
-                    if (this.lexer.lookAhead(0) == ';') {
-                        this.lexer.match(';');
-                    }
-                }
-
-                // Parse remote-tag (if present)
-                if (this.lexer.lookAhead(0) == 'r') {
-                    this.lexer.match('r');
-                    this.lexer.match('=');
-                    targetDialog.setRemoteTag(this.lexer.quotedString());
-                }
+            // Parse parameters (including local and remote tags)
+            while (this.lexer.lookAhead(0) == ';') {
+                this.lexer.consume(1); // Consume the semicolon
+                this.lexer.SPorHT();
+                System.out.println("parameter='" + this.lexer.getBuffer() + "'");
```

Comment:
Let's get rid of System.out and replace with logger.trace but let's comment it in my opinion

Reply:
done